mobile/src/screens/changePin.js

```diff
@@ -0,0 +1,177 @@
+  }
+
+  componentDidMount() {
+    const action = this.props.navigation.getParam('new')
+    this.setState({
+      newAction: action
+    })
+  }
+
+  async handleChange(pin) {
+    // Validate that the pin is numeric
+    if (pin.length && isNaN(pin)) return
+
+    await this.setState({ pin, isRetry: false })
+
+    if (this.state.pin.length === this.pinLength) {
+      if (!this.state.oldPin && this.props.settings.pin === this.state.pin) {
+        // Proceed to verify step, copy value to oldPin
+        this.setState({
+          pin: '',
+          oldPin: this.state.pin,
+          isRetry: false
+        })
+      } else {
+        if (this.props.settings.pin === this.state.oldPin) {
+          // Pin was verified
+          this.props.setPin(this.state.pin)
+          this.props.navigation.goBack()
+        } else {
+          // Pin was incorrect, reset state and try again
+          this.setState({
+            isRetry: true,
+            pin: '',
+            oldPin: null
+          })
+        }
+      }
+    }
+  }
+
+  async handleCreate(pin) {
+    // Validate that the pin is numeric
+    if (pin.length && isNaN(pin)) return
+
+    await this.setState({ pin, isRetry: false })
+
+    if (this.state.pin.length === this.pinLength) {
+      if (!this.state.oldPin) {
+        // Proceed to verify step, copy value to oldPin
+        this.setState({
+          pin: '',
+          oldPin: this.state.pin,
+          isRetry: false
+        })
+      } else {
+        if (this.state.pin === this.state.oldPin) {
+          // Pin was verified
+          this.props.setPin(this.state.pin)
+          this.props.navigation.goBack()
+        } else {
+          // Pin was incorrect, reset state and try again
+          this.setState({
+            isRetry: true,
+            pin: '',
+            oldPin: null
+          })
+        }
+      }
+    }
+  }
+
+  render() {
+    let title = this.state.oldPin
+      ? fbt('Enter your new PIN', 'PinScreen.enterNewPinCode')
+      : fbt('Enter your old PIN', 'PinScreen.enterOldPinCode')
+
+    if (this.state.newAction) {
+      title = this.state.oldPin
+        ? fbt('Re-enter Pin Code', 'PinScreen.reenterPinCode')
+        : fbt('Create a Pin Code', 'PinScreen.createPinCode')
+    }
+    return (
+      <KeyboardAvoidingView
+        style={{ flex: 1 }}
+        behavior={'padding'}
+        keyboardVerticalOffset={Platform.OS === 'android' ? 40 : 0}
+      >
+        <SafeAreaView style={{ flex: 1 }}>
+          <ScrollView
+            contentContainerStyle={[styles.content, styles.greyBackground]}
+            keyboardShouldPersistTaps={'always'}
+          >
+            <View style={styles.container}>
+              <Text style={styles.subtitle}>{title}</Text>
+              {this.state.isRetry === true && (
+                <Text style={styles.invalid}>
+                  <fbt desc="PinScreen.pinMatchFailure">
+                    Old pins did not match, try again
```

Comment:
I think this text should just read "Incorrect PIN" @micahalcorn

Reply:
I fixed.